statsd/client.go

```diff
@@ -304,13 +303,24 @@ func NewClient(addr, prefix string) (Statter, error) {
 	return client, nil
 }
 
-// helper method to ensure a dot is added only when necessary
-func dotprefix(prefix string) string {
-	if prefix != "" && !strings.HasPrefix(prefix, ".") {
-		return "." + prefix
+// joinscopes is a helper that ensures we combine scopes with a dot when
+// it's appropriate to do so. Specifically this looks for trailing dots on
+// the existing prefix and leading dots on the new suffix.
+//
+// It returns the new prefix.
+func joinscopes(curPrefix, suffix string) string {
```

Comment:
graphite calls these "path components" in a hierarchy, not scopes. `joinhier` or `joinPathComp` or something similar would be preferable.